### PR TITLE
net/packet: fix filtering of short IPv4 fragments

### LINKED ISSUE #5727: Medium sized IPv4 fragments are dropped
### What is the issue?

IPv4 fragments up to 640 bytes are dropped.

IPv4 fragments less than 80 bytes should be dropped, but larger than 80 bytes should be fine.

### Steps to reproduce

Use an exit node, browse to http://icmpcheck.popcount.org/, observe second test fails.

### Are there any recent changes that introduced the issue?

_No response_

### OS

_No response_

### OS version

_No response_

### Tailscale version

_No response_

### Bug report

_No response_

### PR DESCRIPTION
**Pull request by raggi**
The fragment offset is an octet-offset rather than a byte offset. Extremely short fragments less than 80 bytes should be dropped, however, fragments of less than 640 bytes are ok.

Fixes #5727

Signed-off-by: James Tucker <james@tailscale.com>

### CODE REVIEW

File: net/packet/packet.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ import (
 const unknown = ipproto.Unknown
 
 // RFC1858: prevent overlapping fragment attacks.
-const minFrag = 60 + 20 // max IPv4 header + basic TCP header
+const minFragOct = (60 + 20) / 8 // max IPv4 header + basic TCP header in octets
```
Comment (DentonGentry) [inline net/packet/packet.go:21]: I agree that there is a problem, but this doesn't look right. It says the IPv4+TCP header is 10 octets, which is not true.

Comment (raggi) [reply net/packet/packet.go:21]: True, the terminology bug starts in the RFC. I'll change it to "fragment blocks".

**Comment by bradfitz** [comment]
Let's add a test.

**Review by DentonGentry — APPROVED** [review]

**Comment by raggi** [comment]
Pausing for a moment, reviewing the flags math. I flipped my test case to see it fail without the fix, and it didn't fail - spuriously. I think we're looking at the flags wrong.

**Comment by raggi** [comment]
Ok, I have corrected the mask, but it's bit twiddling - please scrutinize.

File: net/packet/packet_test.go [hunk 2]
```diff
@@ -265,6 +265,59 @@ var udp4ReplyDecode = Parsed{
 	Dst:     mustIPPort("5.6.7.8:123"),
 }
 
+// First TCP fragment of a packet with leading 24 bytes of 'a's
+var tcp4MediumFragmentBuffer = []byte{
+	// ip header up to checksum
```
Comment (bradfitz) [inline net/packet/packet_test.go:270]: nit: IP in comments. and TCP. and few cases below.

File: net/packet/packet.go [hunk 3]
```diff
@@ -152,11 +152,12 @@ func (q *Parsed) decode4(b []byte) {
 	// it as Unknown. We can also treat any subsequent fragment that starts
 	// at such a low offset as Unknown.
 	fragFlags := binary.BigEndian.Uint16(b[6:8])
-	moreFrags := (fragFlags & 0x20) != 0
+	moreFrags := (fragFlags & 0x2000) != 0
```
Comment (bradfitz) [inline net/packet/packet.go:155]: nice find!

Looks right (now) to me!

**Review by bradfitz — APPROVED** [review]

**Pull request merged**